go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -727,6 +767,27 @@ func (bsnc *BaseSecondaryNetworkController) WatchMultiNetworkPolicy() error {
 		return nil
 	}
 
+	if util.IsNetworkSegmentationSupportEnabled() && bsnc.IsPrimaryNetwork() {
+		return nil
+	}
+
+	if bsnc.policyHandler != nil {
+		return nil
+	}
+	handler, err := bsnc.retryNetworkPolicies.WatchResource()
```

Comment:
Don't we need a different handler for NetPol vs MultiNetPol?

Reply:
I see that you initialize them different. However, I would still use separate fields for clarity.

Reply:
yes, done.